frontend/src/utils/hooks/useResponsiveWidth.tsx

```diff
@@ -7,23 +7,29 @@ export function useResponsiveWidth(
   // Initial spec state is set in useEffect when default geo is set
   const ref = useRef<HTMLDivElement>(document.createElement('div'))
 
+  const forceUpdate = () => {
+    setWidth((prevWidth) => prevWidth + 1)
+  }
+
   useEffect(() => {
-    if (ref?.current) {
-      setWidth(ref.current.offsetWidth)
-    }
+    const element = ref.current
 
     const handleResize = () => {
-      if (ref?.current) {
-        setWidth(ref.current.offsetWidth)
+      if (element) {
+        const newWidth = element.offsetWidth || defaultWidth
```

Comment:
should we use `??` instead of `||` for this short circuit to prevent newWidth from setting to `0` if that's the offset width? or do want `0` to be possible?